code/modules/reagents/chemistry/reagents/cat2_medicine_reagents.dm

```diff
@@ -524,7 +524,23 @@
 		to_chat(carbies, span_danger("You feel your burns and bruises healing! It stings like hell!"))
 
 	carbies.add_mood_event("painful_medicine", /datum/mood_event/painful_medicine)
-	if(HAS_TRAIT_FROM(exposed_mob, TRAIT_HUSK, BURN) && carbies.getFireLoss() < UNHUSK_DAMAGE_THRESHOLD && (carbies.reagents.get_reagent_amount(/datum/reagent/medicine/c2/synthflesh) + reac_volume >= SYNTHFLESH_UNHUSK_AMOUNT))
+
+	//don't unhusked non husked mobs
+	if (!HAS_TRAIT_FROM(exposed_mob, TRAIT_HUSK, BURN))
+		return
+
+	//don't try to unhusk mobs above burn damage threshold
+	if (carbies.getFireLoss() > UNHUSK_DAMAGE_THRESHOLD)
+		return
+
+	var/current_volume = carbies.reagents.get_reagent_amount(/datum/reagent/medicine/c2/synthflesh)
+	var/current_purity = carbies.reagents.get_reagent_purity(/datum/reagent/medicine/c2/synthflesh)
```

Comment:
```suggestion
	var/datum/reagent/synthflesh = carbies.reagents.has(/datum/reagent/medicine/c2/synthflesh)
	if(!synthflesh)
		return

	var/current_volume = synthflesh.volume
	var/current_purity = synthflesh.purity
```

Reply:
i modified it to only use one call of has_reagent, but not quite what you suggested. I don't want to return early because of a weird niche interaction that currently exists. 
if synthflesh is splashed on someone, it should unhusk if they have enough of it in the beaker, even if it's not in their blood.